### PR TITLE
avoid interacting with nil mutatedWfs

### PR DESCRIPTION
**Pull request by ckiosidis**
Signed-off-by: Babis Kiosidis <ckiosidis@gmail.com>

```


goroutine 1299 [running]:
github.com/flyteorg/flytepropeller/pkg/controller.(*Propeller).Handle(0xc0010bcf00, {0x29ee0d8, 0xc2188e2720}, {0xc0009b8140, 0x12}, {0xc0009b8153, 0x8})
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/handler.go:251 +0xd76
github.com/flyteorg/flytepropeller/pkg/controller.(*WorkerPool).processNextWorkItem.func1(0xc001024870, 0xc1f8f1ff28, {0x1ff6080?, 0xc000cbeec0})
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/workers.go:88 +0x510
github.com/flyteorg/flytepropeller/pkg/controller.(*WorkerPool).processNextWorkItem(0xc001024870, {0x29ee0d8, 0xc2188e2720})
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/workers.go:99 +0xf1
github.com/flyteorg/flytepropeller/pkg/controller.(*WorkerPool).runWorker(0x29ee0d8?, {0x29ee0d8, 0xc2192d2d80})
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/workers.go:115 +0xbd
github.com/flyteorg/flytepropeller/pkg/controller.(*WorkerPool).Run.func1()
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/workers.go:150 +0x59
created by github.com/flyteorg/flytepropeller/pkg/controller.(*WorkerPool).Run
	/go/pkg/mod/github.com/flyteorg/flytepropeller@v1.1.28/pkg/controller/workers.go:147 +0x285

```
# TL;DR
when there is an error in the mutatedWf, the handler should avoid interacting with the nil obj

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/controller/handler.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 		t := p.metrics.RoundTime.Start(ctx)
 		mutatedWf, err := p.TryMutateWorkflow(ctx, w)
 
-		if wfClosureCrdFields != nil {
+		if wfClosureCrdFields != nil && mutatedWf != nil {
```
Comment (hamersaw) [inline pkg/controller/handler.go:248]: Can we instead just move this down where the `err != nil` and `mutatuedWf == nil` check are already occurring?

```suggestion
		if err != nil {
			// NOTE We are overriding the deepcopy here, as we are essentially ingnoring all mutations
			// We only want to increase failed attempts and discard any other partial changes to the CRD.
			mutatedWf = RecordSystemError(w, err)
			p.metrics.SystemError.Inc(ctx)
		} else if mutatedWf == nil {
			logger.Errorf(ctx, "Should not happen! Mutation resulted in a nil workflow!")
			return nil
		} else {
			if wfClosureCrdFields != nil {
				// strip data populated from WorkflowClosureReference
				w.SubWorkflows, w.Tasks, w.WorkflowSpec = nil, nil, nil
				mutatedWf.SubWorkflows, mutatedWf.Tasks, mutatedWf.WorkflowSpec = nil, nil, nil
			}

			if !w.GetExecutionStatus().IsTerminated() {
				// No updates in the status we detected, we will skip writing to KubeAPI
				if mutatedWf.Status.Equals(&w.Status) {
					logger.Info(ctx, "WF hasn't been updated in this round.")
					t.Stop()
					return nil
				}
			}
			if mutatedWf.GetExecutionStatus().IsTerminated() {
				// If the end result is a terminated workflow, we remove the labels
				// We add a completed label so that we can avoid polling for this workflow
				SetCompletedLabel(mutatedWf, time.Now())
				ResetFinalizers(mutatedWf)
			}
		}
```

**Review by hamersaw — APPROVED** [review]

**Review by hamersaw — DISMISSED** [review]

**Review by hamersaw — APPROVED** [review]
